src/libraries/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.Specific.cs

```diff
@@ -664,6 +664,31 @@ public async Task ClientConnectAsync_With_Cancellation_Throws_Timeout_When_Pipe_
             }
         }
 
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)] // Unix implementation doesn't rely on a timeout and cancellation token when connecting
+        public async Task ClientConnectAsync_Cancel_With_InfiniteTimeout()
+        {
+            string pipeName = PipeStreamConformanceTests.GetUniquePipeName();
+
+            using (var cts = new CancellationTokenSource())
+            using (var server = new NamedPipeServerStream(pipeName, PipeDirection.InOut, 1))
+            using (var firstClient = new NamedPipeClientStream(pipeName))
+            using (var secondClient = new NamedPipeClientStream(pipeName))
+            {
+                var firstConnectionTasks = new Task[]
+                    {
+                        firstClient.ConnectAsync(),
+                        Task.Run(() => server.WaitForConnectionAsync())
```

Comment:
`WaitForConnectionAsync` returns a `Task`, why would we need an additional `Task.Run` here?

```suggestion
                        server.WaitForConnectionAsync()
```

Reply:
You are absolutely right! Removed the redundant `Task.Run()` call